brainvisa/toolboxes/data management/processes/import/ImportMP2RAGE.py

```diff
@@ -0,0 +1,206 @@
+)
+
+
+# for SPM image, return the orientation in the header file
+def isRadio(hdr):
+    s2m = hdr.get('storage_to_memory')
+    if s2m is None:
+        radio = hdr.get('spm_radio_convention')
+    else:
+        radio = (s2m[0] > 0)
+    return radio
+
+
+# to fetch the name of the subject
+def initSubject(self, inp):
+    if inp is not None and isinstance(inp, DiskItem):
+        value = inp.hierarchyAttributes()
+        if value.get("subject", None) is None:
+            value["subject"] = os.path.basename(inp.fullPath()).partition(".")[0]
+    return value
+
+
+# inputs/outputs initialization and management
+def initialization(self):
+    def _orient(self, proc):
+        old = getattr(proc.signature['input_spm_orientation'], 'lastInput', None)
+        if self.input is None \
+            or old is None \
+            or old.fullPath() != self.input.fullPath() \
+            or proc.isDefault('input_spm_orientation'):
+            hide = 1
+            res = 'Not applicable'
+            if self.input is not None:
+                if self.input.format in list(map(getFormat, ('SPM image', 'Series of SPM image'))):
+                    hide = 0
+                    inputAtts = aimsGlobals.aimsVolumeAttributes(self.input)
+                    radio = isRadio(inputAtts)
+                    if radio is not None and radio != 0:
+                        res = 'Radiological'
+                    else:
+                        res = 'Neurological'
+            if hide:
+                proc.signature['input_spm_orientation'].setChoices('Not applicable')
+            else:
+                proc.signature['input_spm_orientation'].setChoices('Neurological',
+                                                                      'Radiological')
+        else:
+            # print 'using old value:', old, self.input_spm_orientation
+            res = None
+        proc.signature['input_spm_orientation'].lastInput = self.input
+        return res
+
+    self.setOptional('inv1', 'inv2', 't1map', 'uni',
+                     'output_inv1', 'output_inv2', 'output_t1map', 'output_uni')
+    # self.linkParameters('input_spm_orientation', 'input', _orient)
+    self.signature['input_spm_orientation'].linkParameterWithNonDefaultValue = 1
+    self.addLink("output_inv1", "inv1", self.initSubject)
+    self.addLink("output_inv2", "inv2", self.initSubject)
+    self.addLink("output_t1map", "t1map", self.initSubject)
+    self.addLink("output_uni", "uni", self.initSubject)
+    self.signature['output_inv1'].browseUserLevel = 3
+    self.signature['output_inv2'].browseUserLevel = 3
+    self.signature['output_t1map'].browseUserLevel = 3
+    self.signature['output_uni'].browseUserLevel = 3
+    self.signature['inv1'].databaseUserLevel = 2
+    self.signature['inv2'].databaseUserLevel = 2
+    self.signature['t1map'].databaseUserLevel = 2
+    self.signature['uni'].databaseUserLevel = 2
+
+
+# process execution
+def execution(self, context):
```

Comment:
For the import of the T1 "classic" it is not done this way, but with `brainvisa.tools.data_management.image_importation` python module, while the FLAIR or rsfMRI imports have remained on the old version, I suppose, @denisri can you tell us ? Should we do a new import process more like the T1 one with the python module ?

Reply:
The T1 import has been implemented in a python library, because it includes a significant amount of code and operations, since T1 import for Morphologist includes several checks, conversions and corrections in the image: the voxel data should be int16, the values range should be in an appropriate range, etc.
For other image types, I don't know what is needed and what are the operations to be performed in addition to copy / format conversion, so I don't know if it is better to define functions in a python library not not.
Now I actually don't know what are the specificities of the new image types and what they are meant for I wonder, especially for MP2rage: said like that, is also T1w image and can be used as a T1 MRI for Morphologist, but I guess you are planning a different use (and thus possibly different preprocessings) for them...
The thing is that we have not been smart when naming the Morphologist input as "T1 MRI" (25 years ago...), which is a way too general name, we should have named it "morphologist T1" or something like this, including the pipeline name in it since some requirements are actually linked with the pipeline.